middle_end/flambda2/types/meet_and_join_new.ml

```diff
@@ -865,6 +865,15 @@ and meet_head_of_kind_naked_immediate env (t1 : TG.head_of_kind_naked_immediate)
     | false, true ->
       meet_with_shape ~rebuild is_int_ty MTC.any_tagged_immediate is_int_side
   in
+  let is_null_immediate ~is_null_ty ~immediates ~is_null_side =
+    let rebuild = TG.Head_of_kind_naked_immediate.create_is_null in
```

Comment:
In #1324, there's a test about `I.Set.is_empty immediate`  that was added for `is_int` (and already exists for `get_tag`), shouldn't we add it here as well?

Reply:
It didn't make sense before #1324, but I've rebased and added the check.